Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -178,7 +178,7 @@ bool Connection::processMessage()
     Vector<AttachmentInfo> attachmentInfo(attachmentCount);
 
     if (attachmentCount) {
-        memcpy(static_cast<void*>(attachmentInfo.data()), messageData, sizeof(AttachmentInfo) * attachmentCount);
+        memcpySpan(asMutableByteSpan(attachmentInfo), m_readBuffer.span().first(attachmentCount));
```

Comment:
```suggestion
        memcpySpan(asMutableByteSpan(attachmentInfo), m_readBuffer.span().first(sizeof(AttachmentInfo) * attachmentCount));
```

Reply:
But how do we advance `m_readerBuffer` past the `messageInfo`? We need to compensate for the `messageData += sizeof(messageInfo)`.

Reply:
You're right, the code still wouldn't be correct, even after Darin's proposal. We should really use `messageData` and make it a span. This code is still doing pointer arithmetic and is thus very much unsafe still.